File: solutions/TypeScript/[0136_Easy] Single Number/singleNumber.ts
function singleNumber(nums: number[]): number {
  const hashMap = {};
  nums.forEach((n) => {
    if (hashMap[n]) {
      hashMap[n]++;
    } else {
      hashMap[n] = 1;
    }
  });

  return [...new Set(nums)].find((n) => hashMap[n] === 1);
}

File: solutions/Java/[0137_Medium] Single Number II/singleNumber.java
class Solution {
  public int singleNumber(int[] nums) {
    int a = 0;
    int b = 0;

    for (int num: nums) {
      int c = (~a & b & num) | (a & ~b & ~num);
      int d = ~a & (b ^ num);
      a = c;
      b = d;
    }

    return b;
  }
}
